src/Literature/Literature.ts

```diff
@@ -1,14 +1,21 @@
+import { isAbsolutePath } from "../Paths/Directory";
+import { FilePath, isFilePath } from "../Paths/FilePath";
+import type { LiteratureName } from "./data/LiteratureNames";
+
 /**
  * Lore / world building literature files that can be found on servers.
  * These files can be read by the player
  */
 export class Literature {
   title: string;
-  fn: string;
+  fn: LiteratureName & FilePath;
   txt: string;
 
-  constructor(title: string, filename: string, txt: string) {
+  constructor(title: string, filename: LiteratureName, txt: string) {
     this.title = title;
+    if (!isFilePath(filename) || !isAbsolutePath(filename)) {
```

Comment:
This is happening enough that it feels like it should maybe be a helper, where you pass a literal string as part of an error message to use.